### PR TITLE
bugfix and some optimize

### PR DESCRIPTION
**Pull request by lec-bit**
1、fix bug in arm Ubuntu compile
2、optimize dependency package and build
3、adapt image boot file

### CODE REVIEW

File: build/docker/kmesh.yaml [hunk 1]
```diff
@@ -26,6 +26,12 @@ spec:
          - name: kube-config-path
            hostPath:
              path: /root/.kube
+         - name: cni
+           hostPath:
+             path: /etc/cni/net.d
+         - name: kmesh-cniplugin
```
Comment (nlgwcy) [inline build/docker/kmesh.yaml:32]: Please define a semantically clear name, example: kmesh-cniplugin-install-path

File: build/docker/kmesh.yaml [hunk 2]
```diff
@@ -26,6 +26,12 @@ spec:
...
+           hostPath:
+             path: /opt/cni/bin
```
Comment (nlgwcy) [inline build/docker/kmesh.yaml:34]: A comment is added to describe the meaning of the path.

File: build/docker/start_kmesh.sh [hunk 3]
```diff
@@ -1,11 +1,7 @@
 #!/bin/sh
 
-lsmod | grep kmesh > /dev/null
-if [ $? -ne 0 ]; then
-        cp kmesh.ko /lib/modules/`uname -r`
-        depmod -a
-        modprobe kmesh
-fi
+bash build.sh
```
Comment (nlgwcy) [inline build/docker/start_kmesh.sh:3]: Why is it changed to build.sh? Is the original ko installation process unnecessary?

File: build/docker/kmesh.yaml [hunk 4]
```diff
@@ -57,6 +63,12 @@ spec:
            - name: kube-config-path
              mountPath: /root/.kube
              readOnly: true
+           - name: cni # cni conflist file
+             mountPath: /etc/cni/net.d
+             readOnly: false
+           - name: kmesh-cniplugin # k8s default cni path
```
Comment (supercharge-xsy) [inline build/docker/kmesh.yaml:69]: comments and field definitions are ambiguous and need to be clarified.

File: kmesh_compile_env_pre.sh [hunk 5]
```diff
@@ -14,11 +14,11 @@ function install_libboundscheck() {
 function dependency_pkg_install() {
     if command -v apt > /dev/null; then
 	    # apt install 
-	    apt-get update && apt-get install -y git make clang libbpf-dev llvm rpm linux-tools-generic protobuf-compiler libprotobuf-dev libprotobuf-c-dev protobuf-c-compiler cmake golang
+	    apt-get update && apt-get install -y make clang libbpf-dev llvm linux-tools-generic protobuf-compiler libprotobuf-dev libprotobuf-c-dev protobuf-c-compiler cmake golang
```
Comment (nlgwcy) [inline kmesh_compile_env_pre.sh:17]: Do I Need to Update Repo Sources in Online Compilation Scenarios?

**Comment by nlgwcy** [comment]
Pls update the title and description of the PR, including the issues resolved in this modification, and how to modify.

**Comment by nlgwcy** [comment]
> Pls update the title and description of the PR, including the issues resolved in this modification, and how to modify.

**Pull request closed without merging**